modules/oe_translation_cdt/modules/oe_translation_cdt_mock/src/Plugin/ServiceMock/ServiceMockBase.php

```diff
@@ -0,0 +1,171 @@
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * The base class for all CDT API mocks.
+ */
+abstract class ServiceMockBase extends PluginBase implements ServiceMockPluginInterface, ContainerFactoryPluginInterface {
+
+  /**
+   * Constructs a ServiceMockBase object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Extension\ModuleExtensionList $moduleExtensionList
+   *   The module extension list.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
+   *   The logger factory.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    protected ModuleExtensionList $moduleExtensionList,
+    protected EntityTypeManagerInterface $entityTypeManager,
+    protected LoggerChannelFactoryInterface $loggerFactory
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('extension.list.module'),
+      $container->get('entity_type.manager'),
+      $container->get('logger.factory')
+    );
+  }
+
+  /**
+   * Gets the endpoint URL.
+   *
+   * @return string
+   *   The endpoint URL to match against the current request.
+   *   All parameters like ":id" will be considered as wildcards.
+   */
+  abstract protected function getEndpointUrl(): string;
+
+  /**
+   * Gets the response from the endpoint.
+   *
+   * @param \Psr\Http\Message\RequestInterface $request
+   *   The request.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   *   The response.
+   */
+  abstract protected function getEndpointResponse(RequestInterface $request): ResponseInterface;
+
+  /**
+   * Determines if the endpoint needs authorization.
+   *
+   * Override it to FALSE for public endpoints.
+   *
+   * @return bool
+   *   TRUE if the authorization token is required, FALSE otherwise.
+   */
+  protected function needsToken(): bool {
+    return TRUE;
+  }
+
+  /**
+   * Gets the request parameters from the request path.
+   *
+   * @param \Psr\Http\Message\RequestInterface $request
+   *   The request.
+   *
+   * @return array
+   *   The request parameters in the associative array.
+   */
+  protected function getRequestParameters(RequestInterface $request): array {
```

Comment:
Some of the endpoints contain parameters in the path (and not the query string).
Like: https://example.com/api/status/:requestyear/:requestnumber